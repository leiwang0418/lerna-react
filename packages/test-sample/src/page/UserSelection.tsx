const UserSelection = () => <div>UserSelection</div>;

export default UserSelection;