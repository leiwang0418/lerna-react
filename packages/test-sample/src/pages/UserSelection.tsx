import React, { useState } from 'react';
import { Container, Paper } from '@material-ui/core';
import { makeStyles } from '@material-ui/core/styles';
import UserSelectionForm from '../components/UserSelectionForm.container';

const useStyles = makeStyles((theme) => ({
	paper: {
		padding: theme.spacing(2)
	}
}));

const UserSelection = () => {
	const classes = useStyles();

	return (
		<Container maxWidth="sm">
			<Paper className={classes.paper}>
				<UserSelectionForm />
			</Paper>
		</Container>
	);
};

export default UserSelection;
