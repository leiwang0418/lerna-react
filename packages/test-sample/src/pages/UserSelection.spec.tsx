import React from 'react';
import { render } from '@testing-library/react';

import UserSelection from './UserSelection';

jest.mock('../components/UserSelectionForm', () => () => <div>UserSelectionForm</div>);

describe('snapshot test with mock', () => {
	it('renders the basic selection page', () => {
		const { container } = render(<UserSelection />);

		expect(container).toMatchSnapshot();
	})
})
