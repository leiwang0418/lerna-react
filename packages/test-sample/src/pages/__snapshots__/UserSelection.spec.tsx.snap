// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot test with mock renders the basic selection page 1`] = `
<div>
  <div
    class="MuiContainer-root MuiContainer-maxWidthSm"
  >
    <div
      class="MuiPaper-root makeStyles-paper-1 MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div>
        UserSelectionForm
      </div>
    </div>
  </div>
</div>
`;
